p/ext.php

```diff
@@ -44,51 +50,39 @@ function is_valid_path($path) {
 	return is_valid_path_extension($path, CORE_EXTENSIONS_PATH) || is_valid_path_extension($path, THIRDPARTY_EXTENSIONS_PATH);
 }
 
+function sendBadRequestResponse(string $message = null) {
+	header('HTTP/1.1 400 Bad Request');
+	die($message);
+}
+
+function sendNotFoundResponse() {
+	header('HTTP/1.1 404 Not Found');
+	die();
+}
+
+if (!isset($_GET['f']) ||
+	!isset($_GET['t'])) {
+	sendBadRequestResponse('Query string is incomplete.');
+}
+
 $file_name = urldecode($_GET['f']);
 $file_type = $_GET['t'];
+if (!in_array($file_type, array_keys(SUPPORTED_TYPES))) {
```

Comment:
```suggestion
if (empty(SUPPORTED_TYPES[$file_type])) {
```
Minor: Take advantage of the fact that we have an associative array

Reply:
Done!